### PR TITLE
feat: payment processing for ethereum fee proxy

### PR DESCRIPTION
**Pull request by vrolland**

### CODE REVIEW

**Review by benjlevesque — APPROVED** [review]

**Review by benjlevesque — CHANGES_REQUESTED** [review]

File: packages/payment-processor/src/payment/eth-fee-proxy.ts [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+ */
+export function encodePayEthFeeProxyRequest(
+  request: ClientTypes.IRequestData,
+  signerOrProvider: providers.Web3Provider | Signer = getProvider(),
+  feeAmountOverride?: BigNumberish,
+): string {
+  validateRequest(request, PaymentTypes.PAYMENT_NETWORK_ID.ETH_FEE_PROXY_CONTRACT);
+  const signer = getSigner(signerOrProvider);
+
+  const proxyAddress = ethereumFeeProxyArtifact.getAddress(request.currencyInfo.network!);
```
Comment (benjlevesque) [inline packages/payment-processor/src/payment/eth-fee-proxy.ts:60]: same as above

File: packages/payment-processor/src/payment/eth-fee-proxy.ts [hunk 2]
```diff
@@ -0,0 +1,129 @@
...
+
+/**
+ * Return the EIP-681 format URL with the transaction to pay an ETH
+ * Warning: this EIP isn't widely used, be sure to test compatibility yourself.
+ *
+ * @param request
+ * @param amount optionally, the amount to pay. Defaults to remaining amount of the request.
+ * @param feeAmountOverride optionally, the fee amount to pay. Defaults to the fee amount of the request.
+ */
+export function _getEthFeeProxyPaymentUrl(
```
Comment (benjlevesque) [inline packages/payment-processor/src/payment/eth-fee-proxy.ts:75]: not necessary, never been used (should be removed from other PNs too, it's not properly tested)

File: packages/payment-processor/src/payment/utils.ts [hunk 3]
```diff
@@ -214,6 +216,30 @@ export function validateErc20FeeProxyRequest(
   }
 }
 
+/**
+ * Validates the amount and fee parameters for an Eth Fee Proxy based request.
+ * @param request to validate
+ * @param feeAmountOverride optionally, the custom fee amount
+ * @param paymentNetwork defaults to ETH Fee Proxy contract
+ */
+export function validateEthFeeProxyRequest(
```
Comment (benjlevesque) [inline packages/payment-processor/src/payment/utils.ts:225]: I feel this method should be in the eth-fee-proxy file

**Review by benjlevesque — APPROVED** [review]

File: packages/payment-processor/src/payment/utils.ts (hunk 3 above)
Comment (vrolland) [reply packages/payment-processor/src/payment/utils.ts:225]: I made it the same way as for erc20. Again a Refacto mission is required here :face_in_clouds:

Comment (vrolland) [reply packages/payment-processor/src/payment/utils.ts:225]: That is a good question. I would be better for this file but it makes the general organization even more messy.

Comment (vrolland) [reply packages/payment-processor/src/payment/utils.ts:225]: I will go for your point of view anyway :)

Comment (benjlevesque) [reply packages/payment-processor/src/payment/utils.ts:225]: you can put in a new file if you prefer. but the utils should not have specificities of PNs

Comment (vrolland) [reply packages/payment-processor/src/payment/utils.ts:225]: I agree, the questions was should I do it now or in a specific PR refactoring everything.

Comment (vrolland) [reply packages/payment-processor/src/payment/utils.ts:225]: I move it to this file as you suggest

**Pull request merged**